lib/msf/core/web_services/json_rpc_exception_handling.rb

```diff
@@ -0,0 +1,34 @@
+require 'msf/core/rpc'
+
+module Msf::WebServices
+  module JsonRpcExceptionHandling
+
+    def self.registered(app)
+      app.set :show_exceptions => false
+
+      app.error { |err|
+        self.request.body.rewind
```

Comment:
rewind is new to me, it doesn't seem to have any common uses in framework other than a few cases writing files, what's the use case for it here?

Reply:
`body` is a `StringIO` which means once it's been read it (behind the scenes) moves a pointer to the end of it's buffer. This means a subsequent read will return `""` because it's reading from end of the buffer to end of the buffer. `rewind` moves the pointer back to the start of the buffer, allowing subsequent reads to return it's entire contents.

It's needed because at some point before this code `body` was read and not rewound. [TL;DR](https://image.tmdb.org/t/p/original/f0oX20YrQEiVPDH9InCQ1d3Cm66.jpg).